OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameChatLogic.cs

```diff
@@ -317,6 +320,27 @@ void AddChatLine(TextNotification chatLine, bool suppressSound)
 				Game.Sound.PlayNotification(modRules, null, "Sounds", chatLineSound, null);
 		}
 
+		public override void Tick()
+		{
+			var chatWasEnabled = chatEnabled;
+			chatEnabled = Game.RunTime >= orderManager.ChatDisabledUntil && orderManager.ChatDisabledUntil != uint.MaxValue;
```

Comment:
Is it possible to only tick this when we're in the lobby chat? (Can we re-use `isMenuChat` from the ctor?)

Reply:
We need this ingame too (testcase: set a longer timeout and start a game before it expires).

Reply:
Hm, I'm not sure if that is actually a good thing, but I guess this is to cover a spectator entering just as the game is starting.

Reply:
Also future hotfixes or features that might want to turn off a client's chat for some other reason (e.g. server moderator muting an abusive user).